cluster/sbin/snappy-nodes.sh

```diff
@@ -385,6 +417,12 @@ if [ -n "${HOSTLIST}" ]; then
         args="$args -J-Dsnappydata.numLeadsOnHost=$(getNumLeadsOnHost "$host")"
       fi
       execute "$@"
+      if [ -n "$isStart" ]; then
+    	count=$(echo $MEMBERS_FILE | grep -o -i "$host" "$MEMBERS_FILE" | wc -l)
```

Comment:
I think number of occurrence of host can be more than 2 also because he may use the host in the path somewhere. If you want to just take the host count then take the first word from each line.
Also it is better to take care of empty lines or comments line explicitly, though I guess for this file comments are not there since it is generated.

Reply:
checking count is 2, since, in a single line of member.txt, there will be two matches of IP address/hostname of a component. Entry happens in member file sequentially. So for example, if on some node 1, both server and lead are running, then by that time lead will start, member file will have already an entry for IP address/hostname of that node 1 after starting the server. It kind of cross-checking if a component with the same IP address already present in member file or not.

Reply:
Yes, checking ofr count 2 may not always work as mentioned by Neeraj.

Also, we need to copy the conf files **before** the process starts, otherwise, those will not be processed.

We should do the copying just before the process is launched on respective nodes (e.g. in execute() function, just before `if [ "$host" != "localhost" ]; then`  block. This could be done by maintaining the hostnames in a array and updating the array with every iteration and avoid copying the files if the hostname is already present in it.

Reply:
Modified as per review comment